src/rgw/rgw_sal.cc

```diff
@@ -293,6 +448,216 @@ int RGWRadosStore::get_bucket(RGWUser& u, const rgw_bucket& b, RGWBucket** bucke
   return 0;
 }
 
+int RGWRadosStore::get_bucket(RGWUser* u, const RGWBucketInfo& i, RGWBucket** bucket)
+{
+  RGWBucket* bp;
+
+  *bucket = nullptr;
+
+  bp = new RGWRadosBucket(this, i, u);
+  if (!bp) {
+    return -ENOMEM;
```

Comment:
new will throw a `std::bad_alloc` exception if it can't allocate memory, so you'd never hit this. if you really want to disable exceptions here, you can use `new(std::nothrow) RGWRadosBucket(...)`